src/riscv-integration.adoc

```diff
@@ -506,10 +506,18 @@ interrupts cause the <<pcc>> to be set to the capability with
 its address incremented by four times the interrupt cause number.
 
 Capabilities written to <<mtvecc>> also include writing the MODE field in
-**mtvecc.address[1:0]**. As a result, a representability and sealing check is
-performed on the capability with the legalized (WARL) MODE field included in
-the address. The tag of the capability written to <<mtvecc>> is cleared if
-either check fails.
+**mtvecc.address[1:0]**, which is a WARL field, meaning that the capability address can
+change on writing. If the MODE field is non-zero after any WARL conversion,
+then the address used for the exception vector (which has **mtvecc.address[1:0]=2'b00**)
+will not match the value read by software from the CSR.
+
+In this situation, the same capability can have more than one address.
+When updating the CSR _all possible_ values of the address must be in the
+<<section_cap_representable_check>>, and the tag cleared if _any_ are not.
+
+The capability is always written using <<SCADDR>> semantics to update the address field,
```

Comment:
I'm not sure how useful it is to support this but couldn't we install a sentry for non-vectored mode? On exception it is unsealed and in Direct mode there is no address modification so that would work.

For vectored, I agree we have to check that cap address with the low bit cleared (since it is set to 1 for vectored) is representable.

Reply:
it's possible, but that would be a spec change......

Reply:
For MODE=DIRECT the address matches the raw value of the capability so there is only one address that can be observed? We should call out that sentries are invalidated even in this case since there is no unsealing on MRET.

Reply:
https://github.com/riscv/riscv-cheri/issues/561